homeassistant/components/xiaomi_miio/diagnostics.py

```diff
@@ -21,18 +21,17 @@
 
 
 async def async_get_config_entry_diagnostics(
-    hass: HomeAssistant, config_entry: ConfigEntry
+    hass: HomeAssistant, config_entry: XiaomiMiioConfigEntry
 ) -> dict[str, Any]:
     """Return diagnostics for a config entry."""
     diagnostics_data: dict[str, Any] = {
         "config_entry": async_redact_data(config_entry.as_dict(), TO_REDACT)
     }
 
-    # not every device uses DataUpdateCoordinator
-    if coordinator := hass.data[DOMAIN][config_entry.entry_id].get(KEY_COORDINATOR):
-        if isinstance(coordinator.data, dict):
-            diagnostics_data["coordinator_data"] = coordinator.data
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
```

Comment:
> # not every device uses DataUpdateCoordinator

As far as I can tell, that statement was wrong:
- either it's a device (flow_type = CONF_DEVICE) in which case the coordinator data could be anything
- or it's a gateway (flow_type = CONF_GATEWAY) in which case this would most likely have blown up (`coordinator` would have been a `dict[str, DataUpdateCoordinator]` which didn't have a `data` attribute)